web/cobrands/fixmystreet/offline_report.js

```diff
@@ -351,7 +351,11 @@ fixmystreet.offlineReporting = (function() {
             $(document).on('click', '.js-continue-draft', function() {
                 var id = parseInt(this.parentNode.parentNode.getAttribute('data-id'), 10);
                 loadDraft(id).then(function(d) {
-                    location.href = "/report/new?restoreDraft=" + id + "&latitude=" + d.latitude + "&longitude=" + d.longitude;
+                    if (!d.longitude || !d.latitude) {
+                        location.href = "/?setDraftLocation=" + id + '&draftName=' + d.title;
```

Comment:
Makes sense, have included `offline_reporting.js` on the mainpage and added a method there for grabbing the draft title. Also moved the code checking for whether there are drafts out of `offline_drafts.js` and into `offline_reporting.js`.

Reply:
Ah, so the reason for the `offline_report.js`/`offline_draft.js` split is to try and keep the front page JS size as small as possible. Including `offline_report.js` on the front page adds an extra 4k (~18k without gzip) of data transfer for something that is faaaiirrly rarely used, relatively speaking, and while this might not seem much in isolation in aggregate it all adds up. Sorry, I think this is my fault for suggesting the use of `loadDraft` without realising it wasn't already available on the front page.

I think putting `getDraftTitle` in `offline_draft.js` and refactoring it to call `idbKeyval.get('draftOfflineReports')` directly would be the best option, what do you think?

Reply:
Cheers, makes sense. Have inlined the logic to grab the draft title into `offline_draft.js`. Will try to be more size conscious going forward.